src/server/game/Entities/Player/Player.h

```diff
@@ -1946,6 +1946,12 @@ class Player : public Unit, public GridObject<Player>
         uint32 GetMaxPersonalArenaRatingRequirement(uint32 minarenaslot) const;
         void SetHonorPoints(uint32 value);
         void SetArenaPoints(uint32 value);
+        
+        // duel health and mana reset methods
+        void SaveHealthBeforeDuel() { healthBeforeDuel = GetHealth(); }
+        void SaveManaBeforeDuel() { manaBeforeDuel = GetPower(POWER_MANA); }
+        void RestoreHealthAfterDuel() { SetHealth(healthBeforeDuel); }
+        void RestoreManaAfterDuel() { SetPower(POWER_MANA, manaBeforeDuel); }
```

Comment:
Maybe we could add a "custom" store in player which is adaptable but not global.
We could use it to replace several other less used optional core or custom stores,
which would also increase the compression factor for unused entries:

```c++
enum class PlayerExtraUINT32Store
{
    STORE_PLAYER_HEALTH_BEFORE_DUEL,
    STORE_PLAYER_MANA_BEFORE_DUEL
};

Optional<uint32> Player::GetCustomStorageEntry(PlayerExtraUINT32Store::STORE_PLAYER_MANA_BEFORE_DUEL)
{
    ...
}

bool Player::ClearCustomStorageEntry(PlayerExtraUINT32Store::STORE_PLAYER_MANA_BEFORE_DUEL)
{
    ...
}
```